src/domain/entities/Project.ts

```diff
@@ -12,9 +13,19 @@ export type ProjectAttrs = {
     dataSets: DataSet[];
     lastUpdated: ISODateString;
     orgsUnits: OrgUnit[];
+    access: AccessData[];
 };
 
 export class Project extends Struct<ProjectAttrs>() {
+    get uniqueAccessCodes() {
+        return _(this.access)
+            .filter(access => access.type === "groups")
+            .map(access => Project.extractCode(access.name))
+            .filter(accessCode => accessCode.length > 0)
```

Comment:
Now that I see this filter over the accessCode, check if it makes sense for extraCode to return "string | undefined", as we know, empty strings to signal "no value" is typically a red flag.